### PR TITLE
Bugfix/issue 504 spurious integer warning

### LINKED ISSUE #504: spurious integer division warnings
We should only have warnings about integer division when dividing two integers.  As is, if I do something like

```
  real c = 1 / 2.0;
```

I get spurious warning messages:

```
Info: Found int division at '/Users/carp/temp2/foo.stan', line 4, column 11 to column 12:
  1 / 2.0
Positive values rounded down, negative values rounded up or down in platform-dependent way.
```

Also, the rounding in C++ is always done by dropping off the fractional part, so the rounding is toward zero.  (The platform-dependent thing was probably also a bug in the original compiler's message.)

### PR DESCRIPTION
**Pull request by rok-cesnovar**
Fixes #504 
Removed int/real from int warning.

The new error message will be:
```
Info: Found int division at 'operators_int.stan', line 10, column 25 to column 30:
  d_int / d_int
Values will be rounded towards zero.
```

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
@bob-carpenter is "Values will be rounded to zero." fine or do we want any more info?

File: src/frontend/Semantic_check.ml [hunk 1]
```diff
@@ -594,13 +594,12 @@ and semantic_check_expression cf ({emeta; expr} : Ast.untyped_expression) :
...
-        | UInt, UReal, Divide | UInt, UInt, Divide ->
+        | UInt, UInt, Divide ->
             Fmt.pr
-              "@[<hov>Info: Found int division at %s:@   @[<hov 2>%a@]@,%s@,@]"
+              "@[<hov>Info: Found int division at %s:@   @[<hov 2>%a\n@]%s@.@]"
               (Location_span.to_string x.emeta.loc)
               Pretty_printing.pp_expression {expr; emeta}
-              "Positive values rounded down, negative values rounded up or \
-               down in platform-dependent way." ;
+              "Values will be rounded to zero." ;
```
Comment (rybern) [inline src/frontend/Semantic_check.ml:602]: Rounded to 0 makes me think the result will be 0, maybe "rounded toward zero"?

Comment (rok-cesnovar) [reply src/frontend/Semantic_check.ml:602]: Oh lol, yes :)

Comment (rok-cesnovar) [reply src/frontend/Semantic_check.ml]: Fixed.

**Pull request merged**